src/hooks/useCheckboxGroup.jsx

```diff
@@ -0,0 +1,176 @@
+import isEqual from 'lodash.isequal';
+import { useCallback, useContext, useEffect, useMemo, useState } from 'react';
+import FormContext from '../FormContext';
+import logger from '../utils/logger';
+
+/**
+ * @name useCheckboxGroup
+ * @description A hook to be used with the useForm hook, allowing you to manage checkboxes with a similar name as a group.
+ *
+ * @author Timothée Simon-Franza
+ *
+ * @param {object}	[context]		The form context provided by useForm, obligatory only if useCheckboxGroup is called inside the same component as useForm.
+ * @param {object}	[defaultValues]	Optional object used to set "checked" by default on some checkboxes.
+ * @param {string}	name			The name to store all checkboxes under in the form inputs references.
+ * @param {object}	[rules]			Validation rules to apply to the checkboxes as a group.
+ */
+const useCheckboxGroup = ({ defaultValues = {}, name: checkboxGroupName, rules = {} }, context) => {
+	const formContext = useContext(FormContext) ?? context;
+
+	if (!formContext) {
+		logger.fatal('useCheckboxGroup has been called outside of a FormProvider (you can give formContext to useCheckboxGroup directly as a parameter)');
+	}
+
+	const {
+		fieldsRef,
+		formStateRef,
+		validateCheckboxGroup,
+		validateOnChange,
+	} = formContext;
+
+	const [fields, setFields] = useState({});
+	const [errors, setErrors] = useState(() => formContext.formStateRef.current.errors[checkboxGroupName] ?? {});
+
+	/**
+	 * @function
+	 * @name registerCheckbox
+	 * @description A callback method used by a checkbox typed input to register itself to its group.
+	 *
+	 * @author Timothée Simon-Franza
+	 *
+	 * @param {object} checkboxRef The referenced input to register.
+	 */
+	const registerCheckbox = useCallback((checkboxRef) => {
+		if (checkboxRef.value) {
+			fieldsRef.current[checkboxGroupName][checkboxRef.value] = checkboxRef;
+		}
+	}, [fieldsRef, checkboxGroupName]);
+
+	/**
+	 * @function
+	 * @name unregisterCheckbox
+	 * @description A callback method used by a checkbox typed input to unregister itself from its parent group and form.
+	 *
+	 * @author Timothée Simon-Franza
+	 *
+	 * @param {string} checkboxValue The name under which the input's ref has been registered.
+	 */
+	const unregisterCheckbox = useCallback((checkboxValue) => {
+		if (fieldsRef.current[checkboxGroupName] && fieldsRef.current[checkboxGroupName][checkboxValue]) {
+			delete fieldsRef.current[checkboxGroupName][checkboxValue];
+
+			// @TODO: implement a way to remove the related field from the formState's errors object when all checkbox have unmounted.
```

Comment:
Is this TODO still relevant ?

Reply:
In a way, yes: I didn't actually implement any ways to do this yet, but I figured I'd wait a bit to see if it is needed, since the useCheckboxGroup hook doesn't allow manual field destruction like the useFieldArray field does.